src/LocalDateTime.php

```diff
@@ -734,20 +695,60 @@ public function toDateTime() : \DateTime
      * Note that the native DateTimeImmutable object supports a precision up to the microsecond,
      * so the nanoseconds are rounded down to the nearest microsecond.
      */
-    public function toDateTimeImmutable() : \DateTimeImmutable
+    public function toDateTimeImmutable(): DateTimeImmutable
     {
-        return \DateTimeImmutable::createFromMutable($this->toDateTime());
+        return DateTimeImmutable::createFromMutable($this->toDateTime());
     }
 
     /**
      * Serializes as a string using {@see LocalDateTime::__toString()}.
      */
-    public function jsonSerialize() : string
+    public function jsonSerialize(): string
     {
         return (string) $this;
     }
 
-    public function __toString() : string
+    /**
+     * Returns a copy of this `LocalDateTime` with the specified period added.
+     *
+     * @param int $hours   The hours to add. May be negative.
+     * @param int $minutes The minutes to add. May be negative.
+     * @param int $seconds The seconds to add. May be negative.
+     * @param int $nanos   The nanos to add. May be negative.
+     * @param int $sign    The sign, validated as `1` to add or `-1` to subtract.
+     *
+     * @return LocalDateTime The combined result.
+     */
+    private function plusWithOverflow(int $hours, int $minutes, int $seconds, int $nanos, int $sign): LocalDateTime
+    {
+        $totDays =
+            intdiv($hours, LocalTime::HOURS_PER_DAY) +
+            intdiv($minutes, LocalTime::MINUTES_PER_DAY) +
+            intdiv($seconds, LocalTime::SECONDS_PER_DAY);
+        $totDays *= $sign;
+
+        $totSeconds =
+            ($seconds % LocalTime::SECONDS_PER_DAY) +
+            ($minutes % LocalTime::MINUTES_PER_DAY) * LocalTime::SECONDS_PER_MINUTE +
+            ($hours % LocalTime::HOURS_PER_DAY) * LocalTime::SECONDS_PER_HOUR;
+
+        $curSoD = $this->time->toSecondOfDay();
+        $totSeconds = $totSeconds * $sign + $curSoD;
+
+        $totNanos = $nanos * $sign + $this->time->getNano();
+        $totSeconds += Math::floorDiv($totNanos, LocalTime::NANOS_PER_SECOND);
+        $newNano = Math::floorMod($totNanos, LocalTime::NANOS_PER_SECOND);
+
+        $totDays += Math::floorDiv($totSeconds, LocalTime::SECONDS_PER_DAY);
+        $newSoD = Math::floorMod($totSeconds, LocalTime::SECONDS_PER_DAY);
+
+        $newDate = $this->date->plusDays($totDays);
+        $newTime = ($newSoD === $curSoD ? $this->time : LocalTime::ofSecondOfDay($newSoD, $newNano));
+
+        return new LocalDateTime($newDate, $newTime);
+    }
+
+    public function __toString(): string
```

Comment:
Can we have the magic methods at the end of the public methods, instead of after private methods? I don't like it this way.

If not possible, second best choice: back to after `__construct()`.

Reply:
Yep, I've added it. However I can only order it like this ` 'construct', 'phpunit', 'method_public', 'magic', 'method_protected', 'method_private'` which means that if magic methods are protected or private they'll be after public methods anyway. Since it's quite rare I guess it's ok.